solr/solrj/src/java/org/apache/solr/common/util/SimpleOrderedMap.java

```diff
@@ -67,4 +69,42 @@ public SimpleOrderedMap<T> clone() {
     newList.addAll(nvPairs);
     return new SimpleOrderedMap<>(newList);
   }
+
+  /**
+   * Returns a shared, empty, and immutable instance of SimpleOrderedMap.
+   * @return empty SimpleOrderedMap (immutable) 
+   */
+  public static SimpleOrderedMap<Object> of() {
+    return EMPTY;
+  }
+
+  /**
+   * Returns an immutable instance of SimpleOrderedMap with a single element.
+   * @return List containing the elements
+   */
+  public static SimpleOrderedMap<Object> of(Object o1) {
+    return new SimpleOrderedMap<>(List.of(o1));
+  }
+  
+  /**
+   * Returns an immutable instance of SimpleOrderedMap with two elements.
+   * @return List containing the elements
+   */
+  public static SimpleOrderedMap<Object> of(Object o1, Object o2) {
+    return new SimpleOrderedMap<>(List.of(o1,o2));
+  }
+  
+  /**
+   * Returns an immutable instance of SimpleOrderedMap with an arbitrary number of elements. 
+   * @return List containing the elements
+   */
+  public static SimpleOrderedMap<Object> of(Object... elements) {
```

Comment:
Map.of has no similar array based argument so I suggest not doing the same here.  Presumably the key and value distinction then becomes less obvious; easier to misuse.